### PR TITLE
251 intensity transforms

### PR DESCRIPTION
**Pull request by Nic-Ma**
Partly fix issue 251.

### Description
This PR implemented below transforms:
1. ShiftIntensity
2. ShiftIntensityd
3. RandShiftIntensity
4. RandShiftIntensityd
5. ScaleIntensity
6. ScaleIntensityd
7. RandScaleIntensity
8. RandScaleIntensityd

Also totally updated:
9. NormalizeIntensity
10. NormalizeIntensityd

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or new feature that would cause existing functionality to change)
- [x] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### CODE REVIEW

File: monai/transforms/transforms.py [hunk 1]
```diff
@@ -619,30 +605,145 @@ def __call__(self, img):
...
     Args:
         subtrahend (ndarray): the amount to subtract by (usually the mean).
         divisor (ndarray): the amount to divide by (usually the standard deviation).
+        nonzero (bool): whether only normalize non-zero values.
+        each_channel (bool): if using calculated mean and std, calculate on each channel separately
+            or calculate on the entire image directly.
     """
 
-    def __init__(self, subtrahend=None, divisor=None):
+    def __init__(self, subtrahend=None, divisor=None, nonzero=False, each_channel=False):
```
Comment (wyli) [inline monai/transforms/transforms.py:723]: each_channel could be channel_wise

**Comment by Nic-Ma** [comment]
Hi @wyli ,

Thanks for your review.
I updated code according to your comments.
And I will resolve conflicts when other PR merged first.
Thanks.

File: monai/transforms/transforms.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/transforms/transforms.py]: updated in the latest patch.
Thanks.

**Comment by Nic-Ma** [comment]
Hi @wyli ,

As the other transform PRs are merged, I resolved all the conflicts and updated this PR.
Could you please help review it again?

**Pull request merged**